src/http_tracker_connection.cpp

```diff
@@ -117,7 +117,7 @@ namespace libtorrent
 		url += "info_hash=";
 		url += escape_string((const char*)&tracker_req().info_hash[0], 20);
 
-		if (tracker_req().kind == tracker_request::announce_request)
+		if (0 == (tracker_req().kind & tracker_request::scrape_request))
```

Comment:
I actually updated that comment a few moments after I made it basically taking back what I said :)
once I realized you wanted to combine i2p with announce. I think it makes sense with flags